src/lib.rs

```diff
@@ -182,20 +182,25 @@ impl<T> Validated<T> {
     pub fn into(self) -> T {
         self.0
     }
+
+    /// Converts from `&Validated<T>` to `Validated<&T>`.
+    #[inline]
+    pub const fn as_ref(&self) -> Validated<&T> {
+        Validated(&self.0)
+    }
 }
 
 impl<T> AsRef<T> for Validated<T> {
     fn as_ref(&self) -> &T {
-        let Self(inner) = self;
-        inner
+        self
```

Comment:
How does this work?

Reply:
Rust automatically inserts derefs and borrows as needed, so this effectively turns into `&**self` - which calls the `Deref` impl.